### PR TITLE
Update REVLib -> master

### PR DESCRIPTION
**Pull request by drorta**
Yoav: The TODO's will be fixed in another PR (#543)

### CODE REVIEW

File: src/main/java/frc/robot/hardware/rev/motors/BrushedSparkMAXMotor.java
Comment (YHerm) [inline src/main/java/frc/robot/hardware/rev/motors/BrushedSparkMAXMotor.java:1]: pick the name of the package: motors or motor and change in rev and phoenix6

File: src/main/java/frc/robot/hardware/rev/motors/SparkMaxMotor.java [hunk 1]
```diff
@@ -65,8 +65,10 @@ public void updateInputs(InputSignal<?>... inputSignals) {
 
 	@Override
 	public void setBrake(boolean brake) {
-		CANSparkBase.IdleMode idleMode = brake ? CANSparkBase.IdleMode.kBrake : CANSparkBase.IdleMode.kCoast;
-		motor.setIdleMode(idleMode);
+		SparkBaseConfig.IdleMode idleMode = brake ? SparkBaseConfig.IdleMode.kBrake : SparkBaseConfig.IdleMode.kCoast;
+
```
Comment (YHerm) [inline src/main/java/frc/robot/hardware/rev/motors/SparkMaxMotor.java:69]: no need space

File: src/main/java/frc/robot/hardware/rev/motors/SparkMaxWrapper.java [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
 import edu.wpi.first.math.geometry.Rotation2d;
 import frc.utils.Conversions;
 
-public class SparkMaxWrapper extends CANSparkMax {
+public class SparkMaxWrapper extends SparkMax {
 
 	public SparkMaxWrapper(SparkMaxDeviceID deviceID) {
 		super(deviceID.id(), deviceID.type());
-		super.restoreFactoryDefaults();
+
```
Comment (YHerm) [inline src/main/java/frc/robot/hardware/rev/motors/SparkMaxWrapper.java:11]: no need space

File: src/main/java/frc/robot/hardware/rev/motors/SparkMaxMotor.java [hunk 3]
```diff
@@ -65,8 +65,9 @@ public void updateInputs(InputSignal<?>... inputSignals) {
 
 	@Override
 	public void setBrake(boolean brake) {
-		CANSparkBase.IdleMode idleMode = brake ? CANSparkBase.IdleMode.kBrake : CANSparkBase.IdleMode.kCoast;
-		motor.setIdleMode(idleMode);
+		SparkBaseConfig.IdleMode idleMode = brake ? SparkBaseConfig.IdleMode.kBrake : SparkBaseConfig.IdleMode.kCoast;
+		// TODO motor.configure()
```
Comment (tnrz135) [inline src/main/java/frc/robot/hardware/rev/motors/SparkMaxMotor.java:69]: TODO?

**Review by tnrz135 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/hardware/rev/motors/SparkMaxMotor.java (hunk 3 above)
Comment (drorta) [reply src/main/java/frc/robot/hardware/rev/motors/SparkMaxMotor.java:69]: https://github.com/GreenBlitz/GB-Robot-Template/pull/543

File: src/main/java/frc/robot/hardware/phoenix6/motors/TalonFXMotor.java [hunk 4]
```diff
@@ -1,4 +1,4 @@
-package frc.robot.hardware.phoenix6.motor;
+package frc.robot.hardware.phoenix6.motors;
```
Comment (NoamRosenberg08) [inline src/main/java/frc/robot/hardware/phoenix6/motors/TalonFXMotor.java:1]: if its a REVLib change, why does the Phoenix6 Lib is also changed?

**Review by NoamRosenberg08 — CHANGES_REQUESTED** [review]

**Review by GilStein1 — APPROVED** [review]
//todo fix

**Review by YHerm — APPROVED** [review]

**Review by hillelv — APPROVED** [review]

**Review by yonnbro — APPROVED** [review]

**Pull request merged**